### DISCUSSION TITLE
Any tips on how to choose an adequate filter to completely remove unphysical values in the temperature field?

### DISCUSSION THREAD

**Discussion by tuanhtrng** [Nek5000/nekRS#423] [Q&A] [answered]
I simulate natural convection flow with NekRS using the model reported in  [Foroozani et al (2021)](https://doi.org/10.1017/jfm.2020.830). The computational domain is a cylindrical block with a height of 3 times the reference length. The heated surface is in the middle section while the rest are insulating walls.

Mesh was graded spanwise to ensure adequate resolution around discontinuous region between the heated and insulating walls. Primarily results showed fluid circulates top half of the domain so coarser mesh was used for the lower bottom. Meshes toward the walls was refined to capture the boundary layer except circular faces at the top and bottom of the block. I might change that in the future.

Post results show the velocity and pressure field looks fine. Nusselt number of flow with Pr 2094 and Ra 1.1e10 did achieve a steady state value experimented in [Kang & Chung (2010)](https://doi.org/10.1016/j.icheatmasstransfer.2010.06.016) despites unphysical values occurred and propagated the temperature field. Maximum time step and temperature filter (hpfrt filter in the TEMPERATURE par block) decided if the field was stable or exploded.

At the moment, I run the the simulation with variable dt at 1.0e-2 max and the CFL target is 0.4. The subCyclingSteps is 2 but I am scared of running higher CFL since unphysical temperature manifests in the domain of longer runs (10000 seconds).

Is there a way to completely filter unphysical values in the temperature field?

**Comment by stgeke** [comment, marked as answer]
You can try to use AVM. It will help to reduce possible over- and undershoots at the expense of additional diffusion.

**Reply by tuanhtrng** [reply]
I tried and played with AVM as reported in [Lu et al (2019)](https://linkinghub.elsevier.com/retrieve/pii/S1631073X1930158X). The scalar downstream looks much better but small wiggles still exist upstream. Is it Ok to live with them?
